code/src/components/MessageInput.js

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+
+import Button from './Button';
+
+const MessageInput = ({ onMessageChange }) => {
+  const [newMessage, setNewMessage] = useState('');
+
+  //function to post new message to API
+  const handleSubmit = event => {
+    event.preventDefault();
+    onMessageChange(newMessage);
+    setNewMessage('');
+  };
+
+  return (
+    <form onSubmit={handleSubmit} className="Form">
+      <label className="Form__label">
+        <h1 tabIndex="0" className="Form__header">
+          What's making you happy right now?
+        </h1>
+        <textarea
+          className="Form__textarea"
+          rows="4"
+          maxLength="140"
+          value={newMessage}
+          onChange={event => setNewMessage(event.target.value)}
+          placeholder="Write your happy thought..."
+        ></textarea>
+      </label>
+      <p tabIndex="0" className="Form__text">
+        <span
+          className={
+            newMessage.length < 6 || newMessage.length > 140
+              ? 'Form__text--red'
+              : 'Form__text--green'
```

Comment:
A fun thing here could have been if the text went back to red when being 140 (well, 0 in your case on the counter) - since you cannot type anymore in your form (which was neat too!)